CartaPlugin.php

```diff
@@ -17,142 +17,148 @@
+            $elementSet = get_db()->getTable('ElementSet')->findBySql("name = '_carta_version'");
+        }
+        
+        if ((int) $elementSet[0]->description <= 0) {
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN geo_image_olverlays LONGTEXT NOT NULL");
+        }
+        if ((int) $elementSet[0]->description <= 1) {
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN show_sidebar BOOLEAN NOT NULL DEFAULT TRUE");
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN show_minimap BOOLEAN NOT NULL DEFAULT TRUE");
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN show_measure BOOLEAN NOT NULL DEFAULT TRUE");
+        }
+        if ((int) $elementSet[0]->description <= 2) {
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN show_legend BOOLEAN NOT NULL DEFAULT FALSE");
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN legend_content longtext NOT NULL");
+        }
+        if ((int) $elementSet[0]->description <= 3) {
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN latitude varchar(100) NOT NULL DEFAULT '0'");
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN longitude varchar(100) NOT NULL DEFAULT '0'");
+        }
+        if ((int) $elementSet[0]->description <= 4) {
+            $db->query("ALTER TABLE `$db->CartaLayer` ALTER COLUMN `name` varchar(300) NOT NULL");
+            $db->query("ALTER TABLE `$db->CartaLayer` ALTER COLUMN `url` varchar(300) NOT NULL");
+            $db->query("ALTER TABLE `$db->CartaLayer` ALTER COLUMN `key` varchar(300) NOT NULL");
+            $db->query("ALTER TABLE `$db->CartaLayer` ALTER COLUMN `accesstoken` varchar(300) NOT NULL");
+            $db->query("ALTER TABLE `$db->CartaLayer` ALTER COLUMN `attribution` varchar(300) NOT NULL");
+        }
+        if ((int) $elementSet[0]->description <= 5) {
+            $db->query("ALTER TABLE `$db->Carta` ADD COLUMN show_cluster BOOLEAN NOT NULL DEFAULT FALSE");
+        }
+        
+        $db->query("UPDATE `$db->ElementSets` SET description = '{$CurrentVersion}' WHERE name = '_carta_version'");
+        add_shortcode('carta', array($this,'cartaShortcode'));
     }
     
-    public function cartaShortcode($args){        
-       include("shortcode.php");
+    public function cartaShortcode($args)
+    {
+        include("shortcode.php");
     }
 
 
     
-    public function hookPublicHead($args){
+    public function hookPublicHead($args)
+    {
     }
 
 
 
-     public function hookAdminHead($args)
+    public function hookAdminHead($args)
     {
-
     }
 
 
-    public function hookPublicItemsShow($args){
-
+    public function hookPublicItemsShow($args)
+    {
         $item = $args['item'];
-        $id = $item->id; 
+        $id = $item->id;
 
 
         $cartaItem = get_db()->getTable('CartaItem')->getByItemId($item->id);
            
            
-        if(count($cartaItem) > 0){
+        if (is_array($cartaItem) && count($cartaItem) > 0) {
```

Comment:
The fix is here: check if $cartaItem is an array

Reply:
Thank you!

Reply:
It is not currently an array (reads as line 122). 
Do I replace line 122 with line 125? 
Or do I add "if (is_array($cartaItem) && count($cartaItem) > 0) {" to the existing CartaPlugin.php file?

Reply:
I think either should work

Reply:
Thanks! I replaced it and the error message disappeared.

Reply:
Great!